app/controllers/organizations/invitations_controller.rb

```diff
@@ -15,11 +15,13 @@ def create
     authorize! StaffAccount, context: {organization: Current.organization},
       with: Organizations::InvitationPolicy
 
-    @user = User.new(user_params.merge(password: SecureRandom.hex(8)).except(:roles))
+    @user = User.new(
+      user_params.merge(password: SecureRandom.hex(8)).except(:roles)
+    )
+    @user.add_role(user_params[:roles], Current.organization)
     @user.staff_account = StaffAccount.new
 
     if @user.save
-      @user.add_role(user_params[:roles], Current.organization)
```

Comment:
I moved the add_role call to before the `save` call. That way, the role is associated with the user when the user validations happen. Plus, it also means if there is a validation error with adding the role, it will raise before the `User` is persisted.